Q1.java

```diff
@@ -0,0 +1,75 @@
+class Q1
+{
+
+	//TESTING - still working on using JUinit - will update as soon as I have used it successfully 
+	public static void main(String[] args)
+	{
+		boolean[] row1 = new boolean[]{false, true, false, true};
+		boolean[] row2 = new boolean[]{true, true, false, false};
+		boolean[] row3 = new boolean[]{false, false, true, false};
+		boolean[] row4 = new boolean[]{false, false, true, false};
+
+		boolean[][] grid = new boolean[][]{row1, row2, row3, row4};
+
+		System.out.println(countIslands(grid.length, grid[0].length, grid));
+		
+	}
+
+
+
+
+	public static int countIslands(int rows, int columns, boolean[][] grid)
```

Comment:
Good safety mechanism in the method prototype asking the user to pass in the rows and columns, I like this.